executors/cmdexec/block_volume.go

```diff
@@ -126,8 +126,12 @@ func (s *CmdExecutor) BlockVolumeDestroy(host string, blockHostingVolumeName str
 	}
 
 	if blockVolumeDelete.Result == "FAIL" {
-		if strings.Contains(blockVolumeDelete.ErrMsg, "doesn't exist") &&
-			strings.Contains(blockVolumeDelete.ErrMsg, blockVolumeName) {
+		errHas := func(s string) bool {
+			return strings.Contains(blockVolumeDelete.ErrMsg, s)
+		}
+
+		if (errHas("doesn't exist") && errHas(blockVolumeName)) ||
+			(errHas("does not exist") && errHas(blockHostingVolumeName)) {
```

Comment:
Just a nit: This could have been put like this:

```golang
if (errHas("doesn't exist") || errHas("does not exist")) &&
   errHas(blockHostingVolumeName) {
...
```
which I would find a little more natural. At least shorter. ;-)

Reply:
Not really. The first pair is the text "doesn't exist" and the block volume name and the 2nd pair is the text "does not exist" and the BHV name. I checked the gluster-block source code just to be sure but it never prints the block volume name in that error msg, only the gluster volume (BHV) name.
You could try to convince me that I'm being overly cautios about checking that the error messages refer to the volumes, but I like that extra layer of defense at the moment.

Reply:
Dang... I misread that. Must have been too late. :-)))